client/src/partials/purchase/view/purchase_records.js

```diff
@@ -1,24 +1,21 @@
-angular.module('bhima.controllers')
-.controller('purchaseRecords', [
-  '$scope',
-  '$q',
-  'connect',
-  function ($scope, $q, connect) {
-    var session = $scope.session = { purchase_type: 'indirect'};
-    $scope.purchase_filter = {};
+
+var purchaseRecords = function ($q, connect) {
+  var that = this;
```

Comment:
For controllers, the variable name `vm` is recommended by the angular style guide (see [here](https://github.com/johnpapa/angular-styleguide#style-y032)).  `vm` stand for view-model (since the controller binds the view).  For services, the name `service` is recommended (so that you always know you are in a service).

We can choose a different name, but it would be nice to be consistent.  The [login controller](https://github.com/IMA-WorldHealth/bhima/blob/development/client/src/partials/auth/login.js#L18) already uses `vm` as the local variable binding `this`.